### PR TITLE
Feature/frem/either

### PR DESCRIPTION
**Pull request by fremartini**
closing #94

### CODE REVIEW

File: lib/widgets/pages/home_page.dart [hunk 1]
```diff
@@ -34,7 +34,12 @@ class _HomePageState extends State<HomePage> {
...
     try {
-      config = await _configRepository.getAppConfig();
+      final either = await _configRepository.getAppConfig();
+      if (either.success) {
+        config = either.right;
+      } else {
+        config = null;
+      }
     } catch (e) {
       config = null;
```
Comment (jonasanker) [inline lib/widgets/pages/home_page.dart:44]: Do we still need the try catch? Can we avoid to have a null config?

File: lib/utils/either.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
+abstract class Either<L, R> {
+  const Either();
+
+  bool get success => this is Right<L, R>;
+  bool get failure => this is Left<L, R>;
```
Comment (jonasanker) [inline lib/utils/either.dart:5]: To follow the usual implementation pattern of Either, I suggest these should be called `isLeft` and `isRight`

**Review by jonasanker — CHANGES_REQUESTED** [review]
Minor comments on `Either`

File: lib/widgets/pages/home_page.dart (hunk 1 above)
Comment (fremartini) [reply lib/widgets/pages/home_page.dart:44]: This code snippet will be removed in another PR which is why i did not bother to fix it :)

**Review by jonasanker — APPROVED** [review]
Approved ✅

**Pull request merged**